test/test_maskedtensor.py

```diff
@@ -238,8 +237,276 @@ def test_masks_match(self, fn_name):
+        mt.amax().backward()
+        _compare_mts(mt.grad, MaskedTensor(torch.tensor([[0.0, 0, 0], [0, 1, 0]]), m))
+
+    def test_amin(self):
+        d = torch.tensor([[0, 1, 3, -3], [3, -4, 1.0, 3]])
+        m = torch.tensor([[True, False, False, True], [False, True, False, True]])
+        mt = MaskedTensor(d, m)
+        _compare_mts(MaskedTensor(torch.tensor(-4.0), torch.tensor(True)), mt.amin())
+        _compare_mts(
+            MaskedTensor(
+                torch.tensor([0.0, -4.0, 1.0, -3]),
+                torch.tensor([True, True, False, True]),
+            ),
+            mt.amin(dim=0),
+        )
+
+    def test_amin_grad(self):
+        d = torch.tensor([[0, 1, 2], [3, 4, 5.0]])
+        m = torch.tensor([[True, False, False], [False, True, False]])
+        mt = MaskedTensor(d, m, requires_grad=True)
+        mt.amin().backward()
+        _compare_mts(mt.grad, MaskedTensor(torch.tensor([[1.0, 0, 0], [0, 0, 0]]), m))
+
+    def test_prod(self):
+        d = torch.tensor([[0, 1, 3, 0.0], [float("nan"), 4, 1.0, 5.0]])
+        m = torch.tensor([[True, False, False, True], [False, True, False, True]])
+        mt = MaskedTensor(d, m)
+        _compare_mts(MaskedTensor(torch.tensor(0.0), torch.tensor(True)), mt.prod())
+        _compare_mts(
+            MaskedTensor(
+                torch.tensor([0.0, 4.0, 1.0, 0.0]),
+                torch.tensor([True, True, False, True]),
+            ),
+            mt.prod(dim=0),
+        )
+
+    def test_prod_grad(self):
+        d = torch.tensor([[2, float("nan"), 2], [3, 4, 5.0]])
+        m = torch.tensor([[True, False, False], [False, True, False]])
+        mt = MaskedTensor(d, m, requires_grad=True)
+        mt.prod().backward()
+        _compare_mts(mt.grad, MaskedTensor(torch.tensor([[4.0, 0, 0], [0, 2, 0]]), m))
+
+    def test_all(self):
+        d = torch.tensor([[True, True, False, False], [False, True, True, True]])
+        m = torch.tensor([[True, False, False, True], [False, True, False, True]])
+        mt = MaskedTensor(d, m)
+        _compare_mts(MaskedTensor(torch.tensor(False), torch.tensor(True)), mt.all())
+        _compare_mts(
+            MaskedTensor(
+                torch.tensor([True, True, True, False]),
+                torch.tensor([True, True, False, True]),
+            ),
+            mt.all(dim=0),
+        )
+
+        m = torch.tensor([[True, False, True, False], [False, True, False, False]])
+        mt = MaskedTensor(d, m)
+        _compare_mts(
+            MaskedTensor(
+                torch.tensor([True, True, False, True]),
+                torch.tensor([True, True, True, False]),
+            ),
+            mt.all(dim=0),
+        )
+
+    def test_all_grad(self):
+        d = torch.tensor([[True, True, False], [False, True, True]])
+        m = torch.tensor([[True, False, False], [False, True, False]])
+        self.assertRaises(RuntimeError, lambda: MaskedTensor(d, m, requires_grad=True))
```

Comment:
Needs an explanation of what's being tested (and ideally a narrower assert)

Reply:
fixed this to be 

```
        msg = "Only Tensors of floating point and complex dtype can require gradients"
        with self.assertRaisesRegex(RuntimeError, msg):
            MaskedTensor(d, m, requires_grad=True)
```

and changed the test name to `test_grad_dtype` to more accurately reflect what's being tested here